apiserver/modelmanager/modelmanager.go

```diff
@@ -460,7 +460,12 @@ func (m *ModelManagerAPI) ListModels(user params.Entity) (params.UserModelList,
 		return result, errors.Trace(err)
 	}
 
-	models, err := m.state.ModelsForUser(userTag)
+	var models []*state.UserModel
+	if m.isAdmin {
```

Comment:
I tend to agree with jam that this doesn't seem quite right. It seems that this PR is changing the semantics of ListModels, at least partially, to "show me the models that this user has access to" instead of "show me the models owned by this user". Both are viable semantics, but we should choose one and stick to it. This semantics of ListModels in this PR are half way between - if I'm admin, it shows me all models (all the models I have access to) but if I'm not, it only shows me all models owned by me, not all the ones that I have access to.

I'd suggest changing the semantics such that if a user is specified, the existing semantics are kept - it lists all models owned by the given user. But if the user is left empty, it tries to list all models that the user has access to. So for admin, that would list all models in the system, but for a user, it would list all models that they have permission to access.

What do you think?